src/main/cpp/blaze_util.cc

```diff
@@ -181,6 +181,10 @@ void SetDebugLog(bool enabled) {
   }
 }
 
+bool IsRunningWithinTest() {
+  return !GetEnv("TEST_TMPDIR").empty();
```

Comment:
Suggestion: cache the value so that only the first call of `IsRunningWithinTest` will be a syscall.

```cpp
  static const bool is_test = !GetEnv("TEST_TMPDIR").empty();
  return is_test;
```

Reply:
Done.

Good idea and neat implementation, thanks!

Reply:
Unfortunately this breaks `//src/test/cpp:startup_options_test`.

Since the code doesn't call this function often, I'll forgo caching.